### PR TITLE
Msteams webhook

### PR DESCRIPTION
**Pull request by jjh74**
msteams webhook implementation. (This was briefly discussed in https://github.com/alerta/alerta-contrib/issues/280)

### CODE REVIEW

**Comment by jjh74** [comment]
@satterly can you take a look at this PR, AFAIK travis ci fails because the test can't create a test alert in db. Are alerta-contrib tests able to create alerts in db ? If not I can remove the test from PR.

File: webhooks/msteams/alerta_msteamswebhook.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def incoming(self, query_string, payload):
+        # Note: This doesn't validate MS Teams Authorization: Bearer JWT
+        # instead we're relying on alerta to validate X-API-Key header
+
+        action = payload.get('action', 'missing')
+        if action not in ( 'ack', 'close', 'blackout' ):
+            resp = make_response(jsonify(status='error', message='Invalid action'), 400)
+            return resp
+
+        if action in ( 'ack', 'close' ):
```
Comment (satterly) [inline webhooks/msteams/alerta_msteamswebhook.py]: This should be a list not a tuple.
ie `if action in ['ack', 'close']:`

Comment (jjh74) [reply webhooks/msteams/alerta_msteamswebhook.py]: I changed tuples to lists and commented out tests that need mocking (at the moment I don't know how to mock the required Alert.find_by_id, alert.set_status and blackout.create).

**Comment by satterly** [comment]
We can fix the tests in another PR. 👍

**Pull request merged**